benches/fibonacci.rs

```diff
@@ -0,0 +1,182 @@
+use std::{sync::Arc, time::Duration};
+
+use criterion::{
+    black_box, criterion_group, criterion_main, measurement, BatchSize, BenchmarkGroup,
+    BenchmarkId, Criterion, SamplingMode,
+};
+
+use lurk::{
+    eval::{
+        empty_sym_env,
+        lang::{Coproc, Lang},
+        Evaluator,
+    },
+    field::LurkField,
+    proof::nova::NovaProver,
+    proof::Prover,
+    ptr::Ptr,
+    public_parameters::public_params,
+    store::Store,
+};
+
+const DEFAULT_REDUCTION_COUNT: usize = 100;
+fn fib<F: LurkField>(store: &mut Store<F>, a: u64) -> Ptr<F> {
+    let program = format!(
+        r#"
+(let ((fib (lambda (target)
+              (letrec ((next (lambda (a b target)
+                               (if (= 0 target)
+                                     a
+                                     (next b
+                                           (+ a b)
+                                           (- target 1))))))
+                (next 0 1 target)))))
+  (fib {a}))
+"#
+    );
+
+    store.read(&program).unwrap()
+}
+
+fn fibo_total<M: measurement::Measurement>(name: &str, iterations: u64, c: &mut BenchmarkGroup<M>) {
+    let limit: usize = 10_000_000_000;
+    let lang_vesta = Lang::<pasta_curves::Fq, Coproc<pasta_curves::Fq>>::new();
```

Comment:
Please, please don't propagate this mislabeling. It's going to cause confusion or even serious mistakes at some point. I think if you use the defined aliases, this will be clearer. In this case, `pasta_curves::pallas::Scalar` instead of `pasta_curves::Fq`.